src/rev_ai/generic_api_client.py

```diff
@@ -0,0 +1,161 @@
+from .baseclient import BaseClient
+
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin
+
+
+class GenericApiClient(BaseClient):
```

Comment:
This is the main piece that needs feedback. I'm currently unhappy with how submit is functioning. I could use **kwargs and just create a dictionary from all of that but then users don't get argument hints or named arguments

Reply:
Users shouldn't be using this class directly at all. They should be using one of the superclasses which we will provide them TopicExtractionClient is an example of one such class. 

To better explain why I'm unhappy with the current solution: if you look at TopicExtraction client it only needs to have 1 method which is to create its own submit method with explicit "text" and "json" parameters. All the other methods from GenericApiClient just work without TopicExtractionClient having to do anything at all. It would be nice if submit worked in the same way but I don't think its possible to have both that and keep with Python argument conventions.

In our other SDKs I imagine this generic client will just also take a "Submission" object which contains all the information needed to submit a job and they won't need to know the internals of that. However in Python convention is to not have these submission objects and instead have individual arguments on the function directly. An example of this can be seen in the difference between our async api submit calls for [python](https://github.com/revdotcom/revai-python-sdk/blob/develop/src/rev_ai/apiclient.py#L39) and [node](https://github.com/revdotcom/revai-node-sdk/blob/develop/src/api-client.ts#L93). Node takes an options object while python explicitly lists out every argument

Reply:
My concern here is that trying to pass parameters and make it more Python-ish, we are making it [less Python](https://peps.python.org/pep-0020/) (e.g. `Explicit is better than implicit.`).
Once `_submit_job` does not know what payload parameters should be there, maybe it is not its responsibility? So maybe we can assume that the payload is a single argument and `def _submit_job(self, payload)`?

Reply:
Interesting. I'd be happy to discuss this at length. It seems like a choice between being more explicit and the code being DRY. If I do as you're suggesting well have to duplicate payload construction logic everywhere

Reply:
we can tell this:
1. `_submit_xx` is responsible to submit the provided payload
2. `submit_xx` is responsible to construct a concrete payload and submit it using the `_submit`
3. `_enhance_payload(payload, callback, ...)` is called from `submit_`

Reply:
Discussed offline. We will go with this approach as it is more readable and better follows python standards of being explicit even though it slightly increases complexity and code reuse